.github/workflows/check_files.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+"""Script to limit the number of file changes in a single PR.
+Methodology:
+    Analyses the Pull request to find if the count of file changes in a PR
+    exceeds a pre-defined number 20.
+    Checks for unauthorized file changes based on a list of sensitive files.
+This script encourages contributors to align with project practices,
+reducing the likelihood of unintentional merges into incorrect branches.
+NOTE:
+    This script complies with our Python3 coding and documentation standards.
+    It complies with:
+        1) Pylint
+        2) Pydocstyle
+        3) Pycodestyle
+        4) Flake8
+"""
+
+import sys
+import argparse
+import subprocess
+from collections import namedtuple
+import glob
+
+# Use namedtuple for clarity in return values
+ScriptResult = namedtuple('ScriptResult', ['file_count', 'unauthorized_changes'])
+
+def _count_changed_files(base_branch, pr_branch):
+    """
+    Count the number of changed files between two branches.
+    Args:
+        base_branch (str): The base branch.
+        pr_branch (str): The PR branch.
+    Returns:
+        ScriptResult: Namedtuple containing file_count and unauthorized_changes.
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+    base_branch = f"origin/{base_branch}"
+    pr_branch = f"origin/{pr_branch}"
+
+    command = f"git diff --name-only {base_branch}...{pr_branch} --diff-filter=ACMRT | xargs"
+
+    try:
+        # Run git command to get the list of changed files
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+        )
+        output, error = process.communicate()
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
+
+    changed_files = [file.strip() for file in output.split(' ') if file.strip()]
+    file_count = len(changed_files)
+    return ScriptResult(file_count=file_count, unauthorized_changes=changed_files)
+
+def _check_unauthorized_changes(changed_files, sensitive_files):
+    """
+    Check for unauthorized file changes.
+    Args:
+        changed_files (list): List of changed files.
+        sensitive_files (list): List of sensitive files.
+    Returns:
+        list: Unauthorized changes.
+
+    """
+
+    unauthorized_changes = [file for file in changed_files if any(glob.fnmatch.fnmatch(file, sf) for sf in sensitive_files if sf)]
+    return ScriptResult(len(unauthorized_changes), unauthorized_changes)
```

Comment:
It doesn't make sense to return a named tuple if you can infer one attribute from the other. Just return the files found as a list. I recommended this before for another reason but the code has changed